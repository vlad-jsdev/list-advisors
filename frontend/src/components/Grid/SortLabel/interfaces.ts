import { ReactNode } from "react";

export interface SortLabelProps {
	onSort?: (parameters: { direction?: 'asc' | 'desc' | null, keepOther?: boolean }) => void
	children?: ReactNode
	direction?: 'asc' | 'desc'
}