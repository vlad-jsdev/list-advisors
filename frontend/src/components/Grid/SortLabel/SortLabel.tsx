import Button from '@mui/material/Button';

import SortingIcon from "../SortingIcons/SortingIcons";
import { ComponentType, isValidElement, MouseEvent } from 'react';
import { TableHeaderRow } from '@devexpress/dx-react-grid';

const SortLabel: ComponentType<TableHeaderRow.SortLabelProps> = ({ onSort, children, direction }) => {
	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		if (isValidElement(children) && children?.props?.children === 'Reviews' && onSort) {
			onSort({ direction: direction === 'asc' ? 'desc' : 'asc' });
		}
	};
	return (
		<Button size="small" variant="contained" onClick={handleClick}>
			{children}
			{(direction && <SortingIcon direction={direction} />)}
		</Button>

	);
}

export default SortLabel