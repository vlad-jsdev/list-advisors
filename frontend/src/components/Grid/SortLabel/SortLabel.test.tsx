import React from 'react';
import { render } from '@testing-library/react';
import SortLabel from './SortLabel';
import { ColumnChooser } from '@devexpress/dx-react-grid';

describe('SortLabel', () => {
  it('matches the snapshot with the correct props', () => {
    const onSort = jest.fn();
    const children = <span>Sort By Name</span>;
    const direction = 'asc';

    const { asFragment } = render(
      <SortLabel onSort={onSort} direction={direction} column={ColumnChooser} align={''} disabled={false} getMessage={function (messageKey: string): string {
        throw new Error('Function not implemented.');
      } }>
        {children}
      </SortLabel>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
