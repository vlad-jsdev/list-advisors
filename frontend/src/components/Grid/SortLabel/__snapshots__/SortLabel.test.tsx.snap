// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortLabel matches the snapshot with the correct props 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span>
      Sort By Name
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="upward-icon"
      focusable="false"
      style="font-size: 18px;"
      viewBox="0 0 24 24"
    >
      <path
        d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
